models/particle_net.py

```diff
@@ -150,7 +163,8 @@ def __init__(self, K, channels, with_bn=True, activation='relu', pooling='mean',
 
     def call(self, points, features):
         d = self.batch_distance_matrix_general(points, points)
-        _, indicies = tf.nn.top_k(-d, k=self.K + 1)
+        k = self.K if tf.shape(features)[1] >= 16 else tf.shape(features)[1]
```

Comment:
In general it's worth avoiding hardcoded things, so better to have it like (i hope i got the inequality signs correct):
```suggestion
        k = tf.shape(features)[1]-1 if tf.shape(features)[1] <= self.K else self.K
```